Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,314 @@
+
+### Examples
+
+**Example 1a:** A single qubit allocated and released at the same time as the end of the outer block.
+
+```qsharp
+operation FlipCoin() : Result {
+    using (q = Qubit()) {
+        H(q);
+        return M(q);
+    }
+}
+```
+
+The lifetime of the qubit `q` is the same as the lifetime of the surrounding block.
+A new block is syntactically required, but does not serve any purpose other than increasing the indentation level.
+
+**Example 2a:** Many qubits allocated in a single `using` statement with tuple destructuring.
+
+```qsharp
+operation QubitTuple(n : Int) : Result {
+    using ((a, b, c) = (Qubit[2 * n + 1], Qubit[n], Qubit())) {
+        // ...
+        return M(c);
+    }
+}
+```
+
+Allocating multiple qubits requires either a single `using` statement with tuple destructuring, or multiple nested `using` statements.
+Tuple destructuring is hard to read when many variables are declared or when the qubit initializer expression is long.
+It is hard to tell which tuple item corresponds to which variable name.
+
+**Example 3a:** Many qubits allocated with a separate `using` block for each variable.
+
+```qsharp
+operation NestedBlocks(n : Int) : Result {
+    using (a = Qubit[n]) {
+        using (b = Qubit[2 * n + 1]) {
+            using (c = Qubit()) {
+                // ...
+                return M(c);
+            }
+        }
+    }
+}
+```
+
+Allocating multiple qubits with nested `using` statements makes it clear which qubit initializer expression corresponds to which variable name, but at the cost of many levels of unnecessary indentation.
+
+**Example 4a:** A clean qubit and a borrowed qubit.
+When `using` and `borrowing` statements are mixed, nested blocks are required; tuple destructuring is not possible.
+
+```qsharp
+operation UsingAndBorrowing() : Result {
+    using (a = Qubit()) {
+        borrowing (b = Qubit()) {
+            // ...
+            return M(a);
+        }
+    }
+}
+```
+
+**Example 5a:** A qubit allocated in a `using` block but released before the end of the outer block.
+
+```qsharp
+operation DifferentLifetime() : Result {
+    mutable r = Zero;
+    using (q = Qubit()) {
+        // Apply operations...
+        set r = M(q);
+    }
+    // Classical post-processing...
+    return r;
+}
+```
+
+This example shows the main benefit of the current syntax: it allows the lifetime of a qubit to be shorter than the parent block.
+This syntax is preserved by this proposal with only minor changes.
+
+## Proposed Modification
+
+The new `use` and `borrow` statements behave like the existing `using` and `borrowing` statements with two differences:
+
+1. The block is optional.
+   If no block is provided, the scope of the qubit is implicitly defined to be from the `use` or `borrow` statement until the end of the current block.
+2. Parentheses around the header are not required.
+   For consistency, parentheses are no longer required around any block statement header.
```

Comment:
Could you please explicitly list all statements for which there will be a change? Could you please also clarify whether the intent is to have the parenthesis be permanently optional or whether the intent is to ultimately error on parenthesis that are currently valid?